frontend/src/utilities/__tests__/useWatchSegmentKey.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { useWatchSegmentKey } from '~/utilities/useWatchSegmentKey';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { fetchSegmentKey } from '~/services/segmentKeyService';
+
+jest.mock('~/services/segmentKeyService', () => ({
+  fetchSegmentKey: jest.fn(),
+}));
+
+const fetchSegmentKeyMock = fetchSegmentKey as jest.Mock;
+
+describe('useWatchSegmentKey', () => {
+  it('should fetch segment key successfully', async () => {
+    const segmentKeyMock = {
+      segmentKey: 'KRUhoAIEpWlGuz4sWixae1vAXKKGlD5K',
+    };
+    fetchSegmentKeyMock.mockReturnValue(Promise.resolve(segmentKeyMock));
+
+    const renderResult = testHook(useWatchSegmentKey)();
+    expect(fetchSegmentKeyMock).toHaveBeenCalledTimes(1);
+    expect(renderResult.result.current).toStrictEqual({
```

Comment:
nit: prefer using `hooktoStrictEqual` when we working with a hook result.
```suggestion
  expect(renderResult).hookToStrictEqual({
```